From c238308a2a7a5db151d530afd3a96d55b8d7f029 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Tue, 6 Jun 2023 11:23:34 +0200
Subject: [PATCH 0505/1141] arm64: dts: st: Add PCIe EP on the stm32mp257f-ev
 board

Configure PCIe EP and the ComboPHY with external pad clock on
eval-revB board to be used as a device. Exclusive with pcie_rc
that is the default.
The EP is configured for CommonRef clock, need to deassert CLKREF#.
Note that the HW board needs to be modified to root the PCIE_CLK pins
to PCIE_CLKIN.

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: I8acf6faa28921be9a70b5b3a0550a149e0b2b6ca
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/321969
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -541,6 +541,14 @@
 	};
 };
 
+&pcie_ep {
+	pinctrl-names = "default", "init";
+	pinctrl-0 = <&pcie_pins_a>;
+	pinctrl-1 = <&pcie_init_pins_a>;
+	reset-gpios = <&gpioj 8 GPIO_ACTIVE_LOW>;
+	status = "disabled";
+};
+
 &pcie_rc {
 	pinctrl-names = "default", "init";
 	pinctrl-0 = <&pcie_pins_a>;
